### PR TITLE
Val florian rebased

### PR DESCRIPTION
**Pull request by louen**
This PR is the curated version of Florian's branch and mine.
It is *not* ready for merge yet but it is there to help us compare and figure out the remaining work to integrate our code in the cleanest fashion possible.

### CODE REVIEW

File: Plugins/Animation/src/AnimationComponent.cpp [hunk 1]
```diff
@@ -203,6 +207,17 @@ namespace AnimationPlugin
...
+    {
+        for (auto bd : m_boneDrawables)
+        {
+            if (bd->getRenderObjectIndex() == roIndex){
+                return bd->getBoneIndex();
+            }
+        }
+        return uint(-1);
+    }
+
```
Comment (louen) [inline Plugins/Animation/src/AnimationComponent.cpp]: Isn't this a duplicate ?

File: src/Core/Mesh/HalfEdge.cpp [hunk 2]
```diff
@@ -69,35 +69,140 @@ namespace Ra
...
+                HalfEdgeIdx heIdx = m_triangleToHalfEdge[f];
+                HalfEdge cur_he = m_halfEdgeList[heIdx];
+                for (uint i=0; i<m_faceSize[f]-1; ++i)
+                {
+                    CORE_ASSERT( cur_he.m_leftTriIdx == f, "Inconsistent triangle index" );
+                    cur_he = m_halfEdgeList[cur_he.m_next];
+                }
+                CORE_ASSERT( cur_he.m_leftTriIdx == f, "Inconsistent triangle index" );
+                CORE_ASSERT( cur_he.m_next == heIdx, "We didn't end back where we started" );
             }
```
Comment (louen) [inline src/Core/Mesh/HalfEdge.cpp]: Why has this code changed ?

File: src/Core/Mesh/TriangleMesh.hpp [hunk 3]
```diff
@@ -26,9 +26,14 @@ namespace Ra
...
             inline void append( const TriangleMesh& other );
 
+            /// Triangulates quads
+            inline void triangulate();
+
             VectorArray<Vector3>  m_vertices;
             VectorArray<Vector3>  m_normals;
             VectorArray<Triangle> m_triangles;
+
+            VectorArray<VectorNui>  m_faces;
```
Comment (louen) [inline src/Core/Mesh/TriangleMesh.hpp]: I still believe there should be two separate classes : Triangle Mesh and Polygon Mesh

**Comment by louen** [comment]
First the list of features included in this changelist 
## UI
* Change the main UI to a side tab widget
* Use actions to change the renderer (instead of a drop-down)
* Add a "fit camera" button to the UI
* Add an UI for exporting the task graph and the timings

## Animation
* Change the animation output : now export animation time
* add an UI display of animation time
## Core
* Add a flag to tell RelDeb from Release builds
* Add Catmull-Clark and Loop subdivision algorithms
* Add slerp algorithm
* Add cotan projection to polyline
## Engine
* Add a new "End of frame" signal
* Remove BVH stub from renderer

File: src/Core/Mesh/TriangleMesh.hpp (hunk 3 above)
Comment (MathiasPaulin) [inline src/Core/Mesh/TriangleMesh.hpp]: I agree

**Comment by louen** [comment]
Superseded by PR #203

**Pull request closed without merging**